modules/service/main.tf

```
@@ -1653,7 +1653,7 @@ resource "aws_security_group" "this" {
   name        = var.security_group_use_name_prefix ? null : local.security_group_name
   name_prefix = var.security_group_use_name_prefix ? "${local.security_group_name}-" : null
   description = var.security_group_description
-  vpc_id      = data.aws_subnet.this[0].vpc_id
+  vpc_id      = try(data.aws_subnet.this[0].vpc_id, var.vpc_id)
```

This should be the other way around
```suggestion
  vpc_id      = coalesce(var.vpc_id, data.aws_subnet.this[0].vpc_id)
```

---

no that is incorrect - that will fail